Firebase/Core/FIRComponentContainer.m

```diff
@@ -161,14 +171,16 @@ - (nullable id)instanceForProtocol:(Protocol *)protocol {
 
 - (void)removeAllCachedInstances {
   // Loop through the cache and notify each instance that is a maintainer to clean up after itself.
-  for (id instance in self.cachedInstances.allValues) {
-    if ([instance conformsToProtocol:@protocol(FIRComponentLifecycleMaintainer)] &&
-        [instance respondsToSelector:@selector(appWillBeDeleted:)]) {
-      [instance appWillBeDeleted:self.app];
+  dispatch_sync(_containerQueue, ^{
+    for (id instance in self.cachedInstances.allValues) {
```

Comment:
This mutating while looping would have been an issue before the queue was added.

Reply:
I meant if another thread changed it during the loop, there would be a __NSFastEnumerationMutationHandler crash.

Reply:
Ahhh got it, yes that would have been an issue for sure. I don't think it's super common though since it only happens on `FirebaseApp.delete()`, but definitely an issue.